overlays/nodejs/utils.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+type Diff<T extends string | number | symbol, U extends string | number | symbol> =
```

Comment:
This feel out of place in this package.

Reply:
True.  We coudl put it in the main pulumi package.  That's about the only other place i think they coudl live.

Reply:
I think if we need it here we should include it here.  We could just include these immediately in the one place they are being used instead of in `utils`.

I am curious though, what is the error reporting, documentation and intellisense experience here?  I'm tempted to avoid doing this magic and just manually make a copy of all the fields in the struct so that we are in control of documentation and user experience.  Ideally we would include some validation that we haven't "missed" new fields that get added.  But if we could do that validation without having it impact the user facing experience, that is likely preferable (albeit a bit duplicative).